packages/vite/src/node/plugins/asset.ts

```diff
@@ -179,7 +179,7 @@ export function assetPlugin(config: ResolvedConfig): Plugin {
 
       id = id.replace(urlRE, '$1').replace(unnededFinalQueryCharRE, '')
       const url = await fileToUrl(id, config, this)
-      return `export default ${JSON.stringify(url)}`
+      return `export default ${JSON.stringify(`${url}?t=${Date.now()}`)}`
```

Comment:
I need to test how this behaves for public files. But I think this currently marked as deprecated to import public files in source code so I would not worry too much about this.

What are you afraid it will break apart from test suite that are using exact match for paths?

Reply:
I would think that projects shouldn't break as having a `?t=` is expected. It still feels like a non-trivial change. Maybe we could move with it during the 4.4 beta and we see if some projects have issues with this.

Reply:
> But I think this currently marked as deprecated to import public files in source code so I would not worry too much about this.

I don't think it's deprecated (see https://vitejs.dev/guide/assets.html#the-public-directory). But nevertheless I don't see much harm if we add the timestamp for them either.

I think however we should only add the timestamp in dev.